src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/IVSTypeScriptTodoCommentDataServiceImplementation.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.TodoComments;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
+{
+    internal readonly struct VSTypeScriptTodoCommentData
+    {
+        public VSTypeScriptTodoCommentData(VSTypeScriptTodoCommentDescriptor descriptor, string message, int position)
+        {
+            Descriptor = descriptor;
+            Message = message;
+            Position = position;
+        }
+
+        public VSTypeScriptTodoCommentDescriptor Descriptor { get; }
+        public string Message { get; }
+        public int Position { get; }
+    }
+
+    internal readonly struct VSTypeScriptTodoCommentDescriptor
+    {
+        internal readonly TodoCommentDescriptor Descriptor;
```

Comment:
why is this internal? Shouldn't it be private?

Reply:
it's internal because *we* (Roslyn) read back from this.  IT's hidden from TS, but not from us.